HeterogeneousCore/MPICore/plugins/api.h

```diff
@@ -96,7 +103,12 @@ class MPIChannel {
   void receiveSerializedProduct(int instance, edm::ObjectWithDict& product) {
     // the expected use case is that the product type corresponds to the actual type,
     // so we access the type with typeOf() instead of dynamicType()
-    receiveSerializedProduct_(instance, product.typeOf().getClass(), product.address());
+    edm::TypeWithDict type = product.typeOf();
+    if (!type.isFundamental()) {
```

Comment:
I would prefer the keyword based syntax:
```suggestion
    if (not type.isFundamental()) {
```

Reply:
(also above)